src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/JavaScriptTests.cs

```diff
@@ -274,28 +279,36 @@ public static IEnumerable<object> ToEnumerable(this JSObject iterrator)
     },
 };
 ");
-            return (JSObject)Runtime.GetGlobalObject(prefix);
+            return (JSObject)factory.Call();
         }
 
         [Fact]
         public static void AddEventListenerWorks () {
-            var temp = new bool[1];
-            var obj = SetupListenerTest("addEventListenerWorks");
-            obj.AddEventListener("test", () => {
-                temp[0] = true;
+            var temp = new bool[2];
+            var obj = SetupListenerTest();
+            obj.AddEventListener("test", (JSObject envt) => {
+                var data = (int)envt.GetObjectProperty("data");
+                temp[data] = true;
             });
-            obj.Invoke("fireEvent", "test");
+            var evnt0 = obj.Invoke("eventFactory", 0);
+            var evnt1 = obj.Invoke("eventFactory", 1);
+            obj.Invoke("fireEvent", "test", evnt0);
```

Comment:
Why are we firing the event once before GC and again after?

Reply:
We are trying to test that managed side doesn't lose strong reference to evnt0 instance. I added comment